### PR TITLE
use the adventure api

### PR DESCRIPTION
**Pull request by l4zs**
- migrate BungeeCord Chat API to Adventure API
- add some functions to literalText
- update some dependencies and plugins

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -3,21 +3,21 @@ import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
...
     `maven-publish`
     signing
 
-    id("org.jetbrains.dokka") version "1.6.0"
-    kotlin("plugin.serialization") version "1.6.0"
+    id("org.jetbrains.dokka") version "1.6.10"
+    kotlin("plugin.serialization") version "1.6.10"
 
-    id("io.papermc.paperweight.userdev") version "1.3.1"
+    id("io.papermc.paperweight.userdev") version "1.3.4-SNAPSHOT"
```
Comment (jakobkmar) [inline build.gradle.kts:20]: no snapshots please

File: src/main/kotlin/net/axay/kspigot/chat/LiteralTextBuilder.kt [hunk 2]
```diff
@@ -18,10 +23,10 @@ import net.md_5.bungee.api.chat.hover.content.Text
 inline fun literalText(
     baseText: String = "",
     builder: LiteralTextBuilder.() -> Unit = { }
-) = LiteralTextBuilder(baseText).apply(builder).build() as TextComponent
+) = LiteralTextBuilder(baseText).apply(builder).build()
 
-class LiteralTextBuilder(val internalText: BaseComponent, ) {
-    constructor(text: String) : this(TextComponent(text))
+class LiteralTextBuilder(val internalText: Component) {
```
Comment (jakobkmar) [inline src/main/kotlin/net/axay/kspigot/chat/LiteralTextBuilder.kt:28]: comment for this class in general: are you sure that the behaviour will stay the same here? the bungeecord text api had some weird special behaviour, which may not be the same with adventure components

File: src/main/kotlin/net/axay/kspigot/items/KSpigotItems.kt [hunk 3]
```diff
@@ -104,9 +113,9 @@ fun ItemMeta.removeFlags(vararg itemFlag: ItemFlag) = removeItemFlags(*itemFlag)
 /**
  * Provides safe access to the items' displayName.
  */
-var ItemMeta.name: String?
-    get() = if (hasDisplayName()) displayName else null
-    set(value) = setDisplayName(if (value == null || value == "") " " else value)
+var ItemMeta.name: Component?
```
Comment (jakobkmar) [inline src/main/kotlin/net/axay/kspigot/items/KSpigotItems.kt:116]: could you introduce a new `var` called `stringName`, but deprecate it instantly?

File: src/main/kotlin/net/axay/kspigot/chat/LiteralTextBuilder.kt (hunk 2 above)
Comment (l4zs) [reply src/main/kotlin/net/axay/kspigot/chat/LiteralTextBuilder.kt:28]: I tested it using all methods, style variables and even nested literalTexts so it should work as inteded

Comment (jakobkmar) [reply src/main/kotlin/net/axay/kspigot/chat/LiteralTextBuilder.kt:28]: alright, nested literal texts are the part which I was most interested in here

**Pull request merged**